### ISSUE TITLE
Bug: Operator(0.0) not hdf5 serializable

### ISSUE THREAD

**Issue by HugoStrand** [TRIQS/triqs#675]
Dear all,

I just hit a bug in the serialization of the operator class. Trying to store a zero operator segfaults. Here is a minimal example:
```python
from pytriqs.operators import Operator
from pytriqs.archive import HDFArchive

H_int = Operator(0.0)

with HDFArchive('H_int.h5', 'w') as arch:
    arch['H_int'] = H_int
```
This crash and burns with:
```
Traceback (most recent call last):
  File "bug_operator.py", line 8, in <module>
    arch['H_int'] = H_int
  File "/Users/hugstr/apps/triqs_unstable/lib/python2.7/site-packages/pytriqs/archive/hdf_archive.py", line 184, in __setitem__
    val.__write_hdf5__(self._group,key)
RuntimeError: .. Error occurred at Fri Feb  1 10:20:02 2019

.. Error in h5 writing of object Operator
.. C++ error was : 
.. Triqs runtime error at /Users/hugstr/dev/triqs/triqs/h5/base.cpp : 49

Cannot set hyperslab
.. Error occurred on node .. C++ trace is : 
    frame #6: 0x00000001056348cc libtriqs.dylib`triqs::h5::dataspace_from_LS(int, bool, unsigned long long const*, unsigned long long const*, unsigned long long const*, unsigned long long const*) + 652
    frame #7: 0x000000010564325d libtriqs.dylib`triqs::h5::h5_write_attribute(long long, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::vector<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >, std::__1::allocator<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > > const&) + 797
    frame #8: 0x0000000105649cd8 libtriqs.dylib`triqs::hilbert_space::h5_write_attribute(long long, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, triqs::hilbert_space::fundamental_operator_set const&) + 776
    frame #9: 0x000000010566eed9 libtriqs.dylib`triqs::operators::h5_write(triqs::h5::group, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, triqs::operators::many_body_operator_generic<triqs::utility::real_or_complex> const&, triqs::hilbert_space::fundamental_operator_set const&) + 1193
    frame #10: 0x0000000105542a7d operators.so`triqs::operators::h5_write(triqs::h5::group, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, triqs::operators::many_body_operator_generic<triqs::utility::real_or_complex> const&) + 221
    frame #11: 0x0000000105542119 operators.so`Operator___write_hdf5__(_object*, _object*) + 265
```

**Comment by krivenko** [comment]
Hi Hugo,

I believe my recent pull request (https://github.com/TRIQS/triqs/pull/677) fixes this issue.

**Comment by Wentzell** [comment]
Fixed by #677
